### PR TITLE
feat: add `ExecutorApp`

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

This pr is to add a new crossApp `ExecutorApp` which is used for executing msg from BSC. 

For more information, please refer to [BEP363](https://github.com/bnb-chain/BEPs/pull/363).

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add new cross chain channel `ExecutorChannel` and corresponding cross app `ExecutorApp`

### CODE REVIEW

File: x/storage/keeper/cross_app_executor.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+		sMsgServer: storageMsgServer,
+		pMsgServer: paymentMsgServer,
+	}
+}
+
+func (app *ExecutorApp) ExecuteSynPackage(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, payload []byte) sdk.ExecuteResult {
+	pack, err := DeserializeSynPackage(payload)
+	if err != nil {
+		app.sKeeper.Logger(ctx).Error("deserialize executor syn package error", "payload", hex.EncodeToString(payload), "error", err.Error())
+		panic("deserialize executor syn package error")
```
Comment (zjubfd) [inline x/storage/keeper/cross_app_executor.go:36]: Panic will case a failAck package which will case gas on bsc/opbnb, is that ok the oracle module do not handle the crash of this channel?

File: x/storage/keeper/cross_app_executor.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+	}
+
+	switch MsgType(msg.Type) {
+	case MsgTypeCreatePaymentAccount:
+		var gnfdMsg paymentmoduletypes.MsgCreatePaymentAccount
+		err = gnfdMsg.Unmarshal(msg.Data)
+		if err != nil {
+			return err
+		}
+		if !gnfdMsg.GetSigners()[0].Equals(msgSender) {
```
Comment (zjubfd) [inline x/storage/keeper/cross_app_executor.go:96]: Ensure `len(gnfdMsg.GetSigners()) == 1`?

File: x/storage/keeper/cross_app_executor.go [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+			return fmt.Errorf("invalid msg sender")
+		}
+		_, err = app.sMsgServer.CopyObject(sdk.WrapSDKContext(ctx), &gnfdMsg)
+	case MsgUpdateObjectInfo:
+		var gnfdMsg types.MsgUpdateObjectInfo
+		err = gnfdMsg.Unmarshal(msg.Data)
+		if err != nil {
+			return err
+		}
+		if !gnfdMsg.GetSigners()[0].Equals(msgSender) {
```
Comment (zjubfd) [inline x/storage/keeper/cross_app_executor.go:196]: every handler has: `if !gnfdMsg.GetSigners()[0].Equals(msgSender) ` check, can we reuse the code?

File: x/storage/keeper/cross_app_executor.go (hunk 1 above)
Comment (Pythonberg1997) [reply x/storage/keeper/cross_app_executor.go:36]: Modifying oracle module is not convenient. I add a recover here.

File: x/storage/keeper/cross_app_executor.go (hunk 2 above)
Comment (Pythonberg1997) [reply x/storage/keeper/cross_app_executor.go:96]: Done

File: x/storage/keeper/cross_app_executor.go (hunk 3 above)
Comment (Pythonberg1997) [reply x/storage/keeper/cross_app_executor.go:196]: Done

File: x/storage/keeper/cross_app_executor.go (hunk 1 above)
Comment (zjubfd) [reply x/storage/keeper/cross_app_executor.go]: This change is meaningless.

File: x/storage/keeper/cross_app_executor.go [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+func (app *ExecutorApp) msgHandler(ctx sdk.Context, msg ExecutorMsg) error {
+	msgSender, err := sdk.AccAddressFromHexUnsafe(msg.Sender.String())
+	if err != nil {
+		return err
+	}
+
+	switch MsgType(msg.Type) {
+	case MsgTypeCreatePaymentAccount:
+		var gnfdMsg paymentmoduletypes.MsgCreatePaymentAccount
+		err = gnfdMsg.Unmarshal(msg.Data)
```
Comment (zjubfd) [inline x/storage/keeper/cross_app_executor.go:109]: Each message should do `BasicCheck`

Comment (Pythonberg1997) [reply x/storage/keeper/cross_app_executor.go:109]: Done

**Review by zjubfd — APPROVED** [review]

**Review by zjubfd — APPROVED** [review]

**Pull request merged**